lib/commons/dom/is-in-tab-order.js

```diff
@@ -0,0 +1,23 @@
+import AbstractVirtualNode from '../../core/base/virtual-node/abstract-virtual-node';
+import { getNodeFromTree } from '../../core/utils';
+import isFocusable from './is-focusable';
+
+/**
+ * Determines if an element is focusable and able to be tabbed to.
+ * @method isInTabOrder
+ * @memberof axe.commons.dom
+ * @instance
+ * @param {HTMLElement} el The HTMLElement
+ * @return {Boolean} The element's tabindex status
+ */
+export default function isInTabOrder(el) {
+  const vNode = el instanceof AbstractVirtualNode ? el : getNodeFromTree(el);
+
+  if (vNode.props.nodeType !== 1) {
+    return false;
+  }
+
+  const tabindex = parseInt(vNode.attr('tabindex', 10));
```

Comment:
Your code will return `true` for hidden elements with a non-negative tabindex. 

```suggestion
  const tabindex = parseInt(vNode.attr('tabindex', 10));
  if (tabindex <= -1) {
    return false; // Elements with negative tabindex are never in the tab order
  }
```